db/test/fns/object_test.js

```diff
@@ -337,4 +337,104 @@ suite(testing.suiteName(), function() {
     const res = await db.fns.get_object_with_upload('nosuch');
     assert.deepEqual(res, []);
   });
+
+  helper.dbTest('delete_object with hashes', async function(db) {
+    const expires = fromNow('1 day');
+    const uploadId = taskcluster.slugid();
+    await insertData([
+      { name: 'object-1', backend_id: 'be', project_id: 'prj', data: {}, expires, upload_id: uploadId },
+    ]);
+    await db.fns.add_object_hashes('object-1', JSON.stringify({ sha4: '1234', sha6: 'abcdef' }));
```

Comment:
I appreciate the use of made-up algorithms in the tests. I was curious if there was a whitelist of algorithms, and this confirms it is whatever the client wants to send, at least at this level.

Reply:
At the DB level there's no limit.  The API level does limit it, mostly to avoid the possibility of the number of hashes for an object growing without bound, or abuse of the field to store other data.